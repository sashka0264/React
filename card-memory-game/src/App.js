import React, { useState } from 'react';
import { Provider, connect } from 'react-redux';
import { initializeAC as initialize, openCardAC as openCard } from './redux/actions';
import store from './redux/store'
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import styles from "./App.module.scss";

const Settings = ({ maxGrids, initializeAC }) => {
  initializeAC(10);
  const [grid, setGrid] = useState({ error: false, helperText: '', value: '' });

  const setGridValue = ({ target: { value } }) => {
    if (!isNaN(value)) {
      setGrid({...grid, value: +value});
    }
  }

  const startGame = () => {
    if (grid.value > maxGrids ) {
      setGrid({...grid, error: true, helperText: `max ${maxGrids}`});
    } else if (grid.value < 2) {
      setGrid({...grid, error: true, helperText: 'min 2'});
    } else {
      setGrid({...grid, error: false, helperText: ''});
      initializeAC(grid.value);
    }
  }

  return (
    <div className={styles.settings}>
      <TextField 
        error={grid.error} 
        helperText={grid.helperText} 
        id="standard-basic" 
        label="Grid size (n x n, max 10)" 
        value={grid.value}
        onChange={setGridValue}
      />

      <div className={styles.start}>
        <Button variant="outlined" color="secondary" onClick={startGame}>
          Start Game
        </Button>
      </div>
    </div>
  )
}

const AppContainer = () => (
  <Provider store={store}>
    <AppWithState />
  </Provider>
);

const App = ({ initialized, initialize, openCard, maxGrids, gridMap }) => {
  
  if (!initialized) {
    return <Settings maxGrids={maxGrids} initializeAC={initialize} />
  }

  const cardClicked = (id, color) => {
    openCard(id, color)
  }
  
  return (
    <div className={styles.app}>
      <div className={styles.cards}>

      
        {gridMap.map((item) => {
          return (
            <div 
              className={styles.card} 
              key={item.id} 
              id={item.id}
              style={{backgroundColor: item.opened ? item.color : 'orange' }}
              onClick={() => cardClicked(item.id, item.color)}
            >
              {item.opened ? '' : '?'}
            </div>
          )
        })}
      </div>
    </div>
  );
}

const mapStateToProps = ({ initialized, maxGrids, gridMap }) => ({
  initialized,
  maxGrids,
  gridMap
});

const AppWithState = connect(mapStateToProps, { initialize, openCard })(App)


export default AppContainer;
