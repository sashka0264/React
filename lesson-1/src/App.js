import React from 'react';

function Car(props) {
  return (
      <div className="cars__item">
          <h3 className="cars__item-title">{props.model}</h3>
          <p className="cars__item-text">Year: <strong>{props.year}</strong></p>
      </div>
  )
}
// Описание структуры нашего компонента

const App = (
  <div className="cars">
      <Car model="Ford Focus" year="2017"/>
      <Car model="Audi A8" year="2015"/>
      <Car model="Mazda 3" year="2010"/> 
  </div>       
)
// Элемент из наших компонентов



export default App;
