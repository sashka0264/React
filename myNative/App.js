import React, {Component, useState, PureComponent} from 'react';
import { Header } from 'react-native-elements';
import { StyleSheet, Text, View, TextInput, Button, ListView, ScrollView, Image} from 'react-native';
import Spinner from 'react-native-loading-spinner-overlay';

export default class App extends PureComponent {

  state = {
    timeData: null,
    count: "1",
    base: "RUB",
    baseIcon: "₽",
    baseSubtitle: "Russian ruble",
    inputValue: "",
    data: [],
    loading: true,
    errorLoading: false,
    error: false
  }

  getData = () => {
    // console.log("get Data3")
    fetch(`https://api.exchangeratesapi.io/latest?base=${this.state.base}`)
      .then(res => res.json())
      .then(res => {
        let arr = [];
        for (let key in res.rates) {
          let icon, subtitle;
          switch(key) {
            case "RUB": 
              icon = "₽";
              subtitle = "Russian ruble";
              break;
            case "CAD": 
              icon = "$";
              subtitle = "Canadian Dollar";
              break;
            case "HKD": 
              icon = "$";
              subtitle = "Hong Kong Dollar";
              break;
            case "ISK": 
              icon = "kr";
              subtitle = "Icelandic Krona";
              break;
            case "PHP": 
              icon = "₱";
              subtitle = "Philippine Peso";
              break;
            case "DKK": 
              icon = "kr";
              subtitle = "Danish Krona";
              break;
            case "HUF": 
              icon = "Ft";
              subtitle = "Hungarian Forint";
              break;
            case "CZK": 
              icon = "Kč";
              subtitle = "Czech koruna";
              break;
            default: 
              icon = "-";
              subtitle = "-";
              break;
          }
          arr = [
            ...arr,
            {name: key, value: (res.rates[key]), id: key, icon, subtitle}
          ]
        }
        this.setState({data: arr, errorLoading: false, error: false, loading: false, timeData: res.date.replace(/-/g, ".")})
      })
      .catch(err => {
        this.setState({error: true, errorLoading: false})
      });
  }

  changeInput = (e) => {
    let x = false, 
      res = e.nativeEvent.text.replace(/[^0-9.]/, "").split("");
    res = res.map((item, i) => {
      if ((item === "." || item === ",") && x === true) {
        return "";
      } else if ((item === "." || item === ",") && x === false) {
        x = true;
      }
      return item;
    });
    res = res.join("");
    this.setState({count: res});
  }

  componentDidCatch() {
    this.setState({error: true})
  }

  componentDidMount() {
    if (!this.state.error) {
      this.getData();
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if ((prevState.base !== this.state.base)) {
      this.getData();
    } 
  }

  
  newBase = (item) => {
    this.setState({base: item.name, baseSubtitle: item.subtitle, baseIcon: item.icon, loading: true})
  }

  onErrorButton = () => {
    this.setState({errorLoading: true});
    this.getData();
  }

  render() {
    if (this.state.error) {
      return (
        <View style={styles.error}>
          <Text style={styles.errorText}>{this.state.errorLoading ? "Connecting..." : "Something Wrong"}</Text>
          <Button onPress={this.onErrorButton} color="rgba(235, 65, 162, 1)" title="Try again"/>
        </View>
      );
    }
    return (
      <View style={styles.block}>
        <Header
          centerComponent={{ text: `ECB, ${this.state.timeData}`, style: {color: '#fff'}}}
          containerStyle={{
            backgroundColor: 'rgba(51, 94, 137, 1)',
            justifyContent: 'space-around',
          }}
        />

      

        <View style={styles.baseTotal}>
          <TextInput style={styles.baseValue} 

            keyboardType={"number-pad"}
            onChange={this.changeInput} 
            value={this.state.count}
          />

          <View style={styles.baseBlock}>
            <Text style={styles.baseBlockTitle}>
              {this.state.base + " "}
              <Text style={styles.baseBlockIcon}>{this.state.baseIcon}</Text>
            </Text>

            <Text style={styles.baseBlockSubtitle}>{this.state.baseSubtitle}</Text>
          </View>

        </View>
 
      
        {/* <Button style={styles.button} onPress={() => createToDo(inputValue)} title="Добавить"/> */}
      
        
          <ScrollView style={styles.scrolls}>
            {
              !this.state.loading  ?
              this.state.data.map(item => {
                return (
                  <View key={item.id} style={styles.listTotal} onTouchEnd={() => this.newBase(item)}>
                    <View style={styles.listBlock}>
                      <Text style={styles.listBlockTitle}>
                        {item.name + " "}
                        <Text style={styles.listBlockIcon}>{item.icon}</Text>
                      </Text>
                      <Text style={styles.listBlockSubtitle}>{item.subtitle}</Text>
                    </View>
                    <Text style={styles.listBlockCount}>{(item.value*this.state.count).toFixed(2)}</Text>
                  </View>
                )
              }) :
              <Spinner 
                style={styles.spinner}
                visible={this.state.loading} 
                textStyle={styles.spinnerTextStyle}
                overlayColor="rgba(51, 94, 137, 1)"
                color="white"
              />
            }
          </ScrollView> 
        
      </View>
    );
  }
}



const styles = StyleSheet.create({

  errorText: {
    color: "white",
    marginBottom: 10
  },
  error: {
    flex: 6,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "rgba(51, 94, 137, 1)"
  },
  listBlockCount: {
    color: "rgba(235, 65, 162, 1)"
  },  
  scrolls: {
    position: "relative"
  },
  spinnerOverlayColor: {
    color: "white"
  },
  spinnerTextStyle: {
    color: '#FFF'
  },
  listBlockSubtitle: {
    color: "gray",
    fontSize: 15
  },
  listBlockTitle: {
    fontSize: 25
  },
  listBlockIcon: {
    color: "gray",
    paddingLeft: 10
  },
  listTotal: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    paddingLeft: 6,
    paddingRight: 17,
    backgroundColor: "white"
  }, 
  listBlock: {
    backgroundColor: "white",
    padding: 10,
  },
  baseTotal: {
    backgroundColor:'#F8F8F8',
    elevation: 5,
    position:'relative'
  },
  baseBlockTitle: {
    fontSize: 25
  },
  baseBlockIcon: {
    color: "gray",
    paddingLeft: 10
  },
  baseBlockSubtitle: {
    color: "gray",
    fontSize: 15
  },
  baseBlock: {
    backgroundColor: "rgba(232, 234, 255, 1)",
    padding: 10,
    paddingLeft: 17,
    paddingRight: 17
  },
  baseValue: {
    fontSize: 65,
    paddingLeft: 10,
    paddingRight: 10,
    marginTop: 20
  },
  text: {
    marginBottom: 10
  },
  block: {
    fontFamily: "san-serif",
    marginTop: 0,
    width: "100%",
    marginLeft: "auto",
    marginRight: "auto",
    marginBottom: 30
  },
  input: {
    borderStyle: "solid",
    borderWidth: 1,
    padding: 5,
    borderColor: "#a6caf0",
    marginBottom: 10,
    borderRadius: 2
  },
  button: {
    padding: 5
  },
  task: {
    backgroundColor: "gray",
    marginTop: 5,
    padding: 10,
    color: "white",
    borderRadius: 2
  },
  tasks: {
    marginTop: 10
  }
});
