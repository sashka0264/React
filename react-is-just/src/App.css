body {
  background-color: #c1c5d6;
  font-family: sans-serif;
  color: #101118;
}

.app-wrapper {
  display: grid;
  grid-template-rows: 70px 1fr;
  grid-template-columns: 2.4fr 9.6fr;
  grid-gap: 10px;
  grid-template-areas: 
  "header header header"
  "nav contentText contentText";
  max-width: 1200px;
  margin-left: auto;
  margin-right: auto;
}

.app-reference {
  grid-area: contentText;
  margin: 0 0 5px 0;
  border-radius: 2px;
  box-shadow: 0 1px 0 0 #d7d8db, 0 0 0 1px #e3e4e8;
  background-color: #ebecf0;
}


