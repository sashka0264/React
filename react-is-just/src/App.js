/* eslint-disable react/prop-types */
import React, {Component} from "react";
import {Route, withRouter} from "react-router-dom";
import {connect} from "react-redux";
import {compose} from "redux";
import {BrowserRouter as Router} from "react-router-dom";
import {store} from "./redux/store";
import {Provider} from "react-redux";
import HeaderContainer from "./components/Header/HeaderContainer";
import Navbar from "./components/Navbar/Navbar";
import ProfileContainer from "./components/Profile/ProfileContainer";
import Dialogs from "./components/Dialogs/DialogsContainer";
import News from "./components/News/News";
import Music from "./components/Music/Music";
import Settings from "./components/Settings/Settings";
import UsersContainer from "./components/Users/UsersContainer";
import LoginPage from "./components/Login/Login";
import {getMeTC, initializeAppTC} from "./redux/actions";
import Spinner from "./components/common/Spinner/Spinner";
import style from "./App.module.css";

const AppContainer = () => {
  return (
    <Provider store={store}>
      <Router>
        <AppWithRouter/>
      </Router>
    </Provider>
  );
};

export default AppContainer;

export class App extends Component {
  componentDidMount() {
    const {initializeAppTC} = this.props;
    initializeAppTC();
  }

  render() {
    const {initialized} = this.props;
    if (!initialized) {
      return <Spinner/>;
    }
    return (
      <div className={style.appWrapper}>
        <HeaderContainer/>
        <Navbar/>
  
        <div className={style.appReference}>
          <Route path="/profile/:userId?" component={ProfileContainer}/>
          <Route path="/dialogs" component={Dialogs}/> 
          <Route path="/news" component={News}/>
          <Route path="/music" component={Music}/>
          <Route path="/settings" component={Settings}/>
          <Route path="/users" component={UsersContainer}/>
          <Route path="/login" component={LoginPage}/>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({global}) => ({
  initialized: global.app.initialized
});


const AppWithRouter = compose(
  withRouter,
  connect(mapStateToProps, {getMeTC, initializeAppTC})
)(App);


