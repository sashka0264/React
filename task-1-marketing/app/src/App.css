.app-inference {
    display: inline-block;
    background-color: gainsboro;
    padding: 12px 0px 15px 25px;
    border-radius: 5px;
    max-height: 430px;
}

.app-output {
    margin-top: 10px;
    border-radius: 5px;
    background: gainsboro;
    padding: 0 10px 0 10px;
}

.app-dates {
    display: flex;
}

.app-software {
    display: flex;
    box-shadow: inset 2px 3px 7px rgba(0,0,0,.35);
    background-color: rgba(255,127,99,.26);
    max-width: 330px;
    margin-top: 20px;
    border-radius: 3px;
}