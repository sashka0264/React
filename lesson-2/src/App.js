import React from 'react';
import logo from './logo.svg';
import './App.css';

const basis = (
  <div>
      <App/>
  </div>
)

const App = () => {
  return (
    <div className="App">
      <ul>
        <li>Какой-то текст</li>
        <li>Какой-то текст2</li>
      </ul>
    </div>
  );
}

export default App;
