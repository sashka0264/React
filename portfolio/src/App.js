import React from 'react';
import './App.css';

const App = () => {
  const images = ["./img/one.jpg", "./img/two.jpg", "./img/free.jpg"]
  Promise.all(images.map(url => {
    return new Promise((resolve, reject) => {
      const img = new Image(400, 300)
      img.src = url
      img.onload = () => resolve(img)
      img.onerror = () => reject(url)
    })
  }))
    .then(res => {
      console.log(res)
    })
  return <div>Some text...</div>
}

export default App;
