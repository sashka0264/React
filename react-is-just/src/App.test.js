import React from "react";
import { shallow } from "enzyme";
import { App } from "./App";
import {createSerializer} from "enzyme-to-json";

expect.addSnapshotSerializer(createSerializer({mode: "deep"}));

describe("App", () => {
  const mockinitializeAppTC = jest.fn(); // создание мока для функции 'mockinitializeAppTC'

  const props = {
    initializeAppTC: mockinitializeAppTC,
    initialized: false
  };

  it("dispatches the 'initializeAppTC()' method it receives from props", () => {
    const AppShallow = shallow(<App {...props} />);
    expect(mockinitializeAppTC).toHaveBeenCalledTimes(1);
    expect(AppShallow).toMatchSnapshot();
  });
});