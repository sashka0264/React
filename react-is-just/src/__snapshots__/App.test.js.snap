// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App dispatches the 'initializeAppTC()' method it receives from props 1`] = `<Spinner />`;
